error in build during " Provision AWS SaaS Boost into your AWS Account"

## :question: General Issue


### The Question
following "getting started guide" 
https://github.com/awslabs/aws-saas-boost/blob/main/docs/getting-started.md

invoking the "sh ./install.sh" command only reports  "Error building installer for SaaS boost"

(see other information below).  How are the awscredentials resolved to fix this error?

### Environment

  - **AWS SaaS Boost Version:**  2.2.0
  - Apache Maven 3.9.1
  - openjdk version "11.0.18" 2023-01-17 LTS
    OpenJDK Runtime Environment Corretto-11.0.18.10.1 (build 11.0.18+10-LTS)
    OpenJDK 64-Bit Server VM Corretto-11.0.18.10.1 (build 11.0.18+10-LTS, mixed mode)
-aws-cli/2.11.1
-git version 2.39.2 (Apple Git-143)

### Other information

Found more detail on error in the surefire-reports:

Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.212 s <<< FAILURE! - in com.amazon.aws.partners.saasfactory.saasboost.clients.RefreshingProfileDefaultCredentialsProviderTest
refreshingCredentialsProviderFindsNewCredentials(com.amazon.aws.partners.saasfactory.saasboost.clients.RefreshingProfileDefaultCredentialsProviderTest)  Time elapsed: 0.128 s  <<< FAILURE!
java.lang.AssertionError: expected:<AwsCredentials(accessKeyId=AKIAIOSFODNN7EXAMPLE)> but was:<AwsCredentials(accessKeyId=

actual accesskeyid deleted from this report

Hello,

I will work through the process using cloud9 later today.  Here are my responses to your questions.  Thank you for your assistance.  

MacBook Pro, 2.8Ghz i7, 16GB, Ventura 13.3.1
Corretta installed:
https://docs.aws.amazon.com/corretto/latest/corretto-11-ug/macos-install.html
Maven installed:  
brew install maven

aws configure list-profiles: default
AWS_PROFILE was not exported and is not set. 

I have the following set up in my zsh shell & .aws/credentials

AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=

redirected output of mvn command in install.sh to a file instead of /dev/null.  Here is the output:

[ERROR] Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.212 s <<< FAILURE! - in com.amazon.aws.partners.saasfactory.saasboost.clients.RefreshingProfileDefaultCredentialsProviderTest
[ERROR] refreshingCredentialsProviderFindsNewCredentials(com.amazon.aws.partners.saasfactory.saasboost.clients.RefreshingProfileDefaultCredentialsProviderTest)  Time elapsed: 0.128 s  <<< FAILURE!
java.lang.AssertionError: expected:<AwsCredentials(accessKeyId=AKIAIOSFODNN7EXAMPLE)> but was:<AwsCredentials(accessKeyId=xxxxxxxxxxxxxxxxxx)>
	at com.amazon.aws.partners.saasfactory.saasboost.clients.RefreshingProfileDefaultCredentialsProviderTest.runUpdatingCredentialsProviderTest(RefreshingProfileDefaultCredentialsProviderTest.java:129)
	at com.amazon.aws.partners.saasfactory.saasboost.clients.RefreshingProfileDefaultCredentialsProviderTest.refreshingCredentialsProviderFindsNewCredentials(RefreshingProfileDefaultCredentialsProviderTest.java:121)

[ERROR] defaultCredentialsProviderBugStillExists(com.amazon.aws.partners.saasfactory.saasboost.clients.RefreshingProfileDefaultCredentialsProviderTest)  Time elapsed: 0.013 s  <<< FAILURE!
java.lang.AssertionError: expected:<AwsCredentials(accessKeyId=AKIAIOSFODNN7EXAMPLE)> but was:<AwsCredentials(accessKeyId=xxxxxxxxxxxxxxxx)>
	at com.amazon.aws.partners.saasfactory.saasboost.clients.RefreshingProfileDefaultCredentialsProviderTest.runUpdatingCredentialsProviderTest(RefreshingProfileDefaultCredentialsProviderTest.java:129)
	at com.amazon.aws.partners.saasfactory.saasboost.clients.RefreshingProfileDefaultCredentialsProviderTest.defaultCredentialsProviderBugStillExists(RefreshingProfileDefaultCredentialsProviderTest.java:107)

Enter a String value for parameter RequiredStringParameter: [ERROR] Failures: 
[ERROR]   RefreshingProfileDefaultCredentialsProviderTest.defaultCredentialsProviderBugStillExists:107->runUpdatingCredentialsProviderTest:129 expected:<AwsCredentials(accessKeyId=AKIAIOSFODNN7EXAMPLE)> but was:<AwsCredentials(accessKeyId=xxxxxxxxxxxxxxxx)>
[ERROR]   RefreshingProfileDefaultCredentialsProviderTest.refreshingCredentialsProviderFindsNewCredentials:121->runUpdatingCredentialsProviderTest:129 expected:<AwsCredentials(accessKeyId=AKIAIOSFODNN7EXAMPLE)> but was:<AwsCredentials(accessKeyId=xxxxxxxxxxxx)>
[ERROR] Tests run: 21, Failures: 2, Errors: 0, Skipped: 0
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project SaaSBoostInstall: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/rod/development/projects/saas/aws-saas-boost/installer/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

--- a/installer/src/test/java/com/amazon/aws/partners/saasfactory/saasboost/clients/RefreshingProfileDefaultCredentialsProviderTest.java
+++ b/installer/src/test/java/com/amazon/aws/partners/saasfactory/saasboost/clients/RefreshingProfileDefaultCredentialsProviderTest.java
@@ -31,11 +31,16 @@ import java.io.File;
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
 
 import static org.junit.Assert.*;
+import static org.junit.Assume.assumeFalse;
 
 public class RefreshingProfileDefaultCredentialsProviderTest {
     private static final Logger LOGGER = LoggerFactory.getLogger(RefreshingProfileDefaultCredentialsProviderTest.class);
+
+    private static final String AWS_ACCESS_KEY_ID = "AWS_ACCESS_KEY_ID";
+    private static final String AWS_SECRET_ACCESS_KEY = "AWS_SECRET_ACCESS_KEY";
 
     private static final String BEFORE_ACCESS_KEY = "AKIAIOSFODNN7EXAMPLE";
     private static final String AFTER_ACCESS_KEY = "AKIAI44QH8DHBEXAMPLE";
@@ -57,12 +62,36 @@ public class RefreshingProfileDefaultCredentialsProviderTest {
 
     private static final String RESOURCES_LOCATION = "src/test/resources/";
 
+    // this is created to hide all AWS credentials from the test environment that are not profile credentials
+    // so that the RefreshingProfileDefaultCredentialsProvider can be tested directly.
     private static NoSystemPropertyCredentialsTestEnvironment cachedSystemProperties;
 
     private String fakeProfileFilename;
 
     public RefreshingProfileDefaultCredentialsProviderTest() {
 
+    }
+
+    /**
+     * Returns whether to skip these tests based on AWS Credentials Environment Variable configuration.
+     * 
+     * As stated in the documentation for the NoSystemPropertyCredentialsTestEnvironment, if environment
+     * variables are configured for this test they will not be hidden and this test cannot verify the
+     * functionality of the RefreshingProfileDefaultCredentialsProvider.
+     * 
+     * In most cases it is acceptable to skip these tests, since if users are running these tests with
+     * the EnvironmentVariableCredentialsProvider that means they will likely be running the Installer
+     * itself with the same configuration, so the the RefreshingProfileDefaultCredentialsProvider will
+     * not be used at all.
+     * 
+     * The only caveat is that any change to the RefreshingProfileDefaultCredentialsProviderTest must
+     * guarantee that this test is run before being accepted for contribution.
+     * 
+     * @return true if AWS_ACCESS_KEY_ID or AWS_SECRET_ACCESS_KEY is configured in environment
+     */
+    private static boolean shouldSkipTests() {
+        Set<String> environmentKeys = System.getenv().keySet();
+        return environmentKeys.contains(AWS_ACCESS_KEY_ID) || environmentKeys.contains(AWS_SECRET_ACCESS_KEY);
     }
 
     @BeforeClass
@@ -124,6 +153,10 @@ public class RefreshingProfileDefaultCredentialsProviderTest {
     private void runUpdatingCredentialsProviderTest(
     AwsCredentialsProvider credentialsProviderToTest,
             boolean expectUpdate) throws IOException {
+        assumeFalse("Skipping test due to configuration of AWS credentials as Environment Variables." 
+                + " To run this test unset the environment variables "
+                + AWS_ACCESS_KEY_ID + " and " + AWS_SECRET_ACCESS_KEY,
+                shouldSkipTests());
         AwsCredentials expectedCredentials = TEST_CREDENTIALS[0];
         for (int i = 1 ; i < TEST_CREDENTIALS.length ; i++) {
             assertEquals(expectedCredentials, credentialsProviderToTest.resolveCredentials());

